thirdeye/thirdeye-spi/src/main/java/org/apache/pinot/thirdeye/detection/performance/PerformanceMetrics.java

```diff
@@ -0,0 +1,136 @@
+    private final String FALSE_POSITIVE = "FALSE_POSITIVE";
+    private final String FALSE_NEGATIVE = "FALSE_NEGATIVE";
+
+    public Builder (List<String> statusClassifications) {
+      statusClassifications.stream()
+          .forEach(classification -> {
+            this.numAnomalies++;
+            if (!NOT_CLASSIFIED.equals(classification)) {
+              this.respondedAnomalies++;
+              switch (classification) {
+                case TRUE_POSITIVE:
+                  this.truePos++;
+                  break;
+                case FALSE_NEGATIVE:
+                  this.falseNeg++;
+                  break;
+                case FALSE_POSITIVE:
+                  this.falsePos++;
+                  break;
+                default:
+                  break;
+              }
+            }
+          });
+    }
+
+    public Builder addTotalAnomalies() {
+      this.includeTotalAnomalies = true;
+      return this;
+    }
+
+    public Builder addResponseRate() {
+      this.includeResponseRate = true;
+      return this;
+    }
+
+    public Builder addPrecision() {
+      this.includePrecision = true;
+      return this;
+    }
+
+    public Builder addRecall() {
+      this.includeRecall = true;
+      return this;
+    }
+
+    public PerformanceMetrics build() {
+      PerformanceMetrics pm = new PerformanceMetrics();
+      if (this.includeTotalAnomalies) {
+        pm.totalAnomalies = buildTotalAnomalies();
+      }
+      if (this.includeResponseRate) {
+        pm.responseRate = buildResponseRate();
+      }
+      if (this.includePrecision) {
+        pm.precision = buildPrecision();
+      }
+      if (this.includeRecall) {
+        pm.recall = buildRecall();
+      }
+      return pm;
+    }
+
+    private PerformanceMetric buildTotalAnomalies() {
+      PerformanceMetric totalAnomalies = new PerformanceMetric();
+      totalAnomalies.setValue((double)this.numAnomalies);
+      totalAnomalies.setType(PerformanceMetricType.COUNT);
+      return totalAnomalies;
+    }
+
+    private PerformanceMetric buildResponseRate() {
+      PerformanceMetric responseRate = new PerformanceMetric();
+      double rate = (double)this.respondedAnomalies / this.numAnomalies * 100;
+      responseRate.setValue(rate);
+      responseRate.setType(PerformanceMetricType.PERCENT);
+      return responseRate;
+    }
+
+    private PerformanceMetric buildPrecision() {
+      PerformanceMetric precision = new PerformanceMetric();
+      double prec = (double)this.truePos / (this.truePos + this.falsePos) * 100;
+      precision.setValue(prec);
+      precision.setType(PerformanceMetricType.PERCENT);
+      return precision;
+    }
+
+    private PerformanceMetric buildRecall() {
+      PerformanceMetric recall = new PerformanceMetric();
+      double rec = (double)this.truePos / (this.truePos + this.falseNeg) * 100;
+      recall.setValue(rec);
+      recall.setType(PerformanceMetricType.PERCENT);
+      return recall;
+    }
```

Comment:
Minor - For improved readability I would suggest defining these functions before the public functions, right below the private constants, so we can immediately distinguish between private and public items of this class.